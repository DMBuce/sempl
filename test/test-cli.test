#!/bin/bash
# test sempl interface
#
# TODO: test options once they're added

export template='{$check}'
export check=OK
export tmpfile=/tmp/sempl.test

echo -n "Template source is stdin: "
echo "$template" | sempl

echo -n "Template source is -: "
echo "$template" | sempl -

echo -n "Template source is arg: "
sempl "$template"

echo -n "Template source is file: "
sempl <(echo "$template")

echo -n "Template dest is stdin: "
echo "$template" | sempl -

echo -n "Template dest is -: "
echo "$template" | sempl - -

echo "Template dest is file, no output expected:"
rm -f "$tmpfile"
sempl "$template" "$tmpfile"

echo -n "File from previous test has output: "
cat "$tmpfile"

echo -n "Template dest can be overwritten: "
check='STILL OK' sempl "$template" "$tmpfile"
cat "$tmpfile"

echo -n "Source and dest can't be the same, error expected: "
sempl "$tmpfile" "$tmpfile"

echo -n "Test \$SEMPL_BRACES: "
SEMPL_BRACES='{{}}' sempl '{{$check}}'

echo -n "Test \$LB: "
sempl '{$LB}$check}'

echo -n "Test \$RB, TODO fix this: "
sempl '{$check{$RB}'

echo -n "Test \$SEMPL_DUMP: "
SEMPL_DUMP=1 sempl "$template"

echo "Test \$SEMPL_DUMP writing to dest, no output expected:"
rm -f "$tmpfile.sh"
SEMPL_DUMP=1 sempl "$template" "$tmpfile"

echo -n "File from previous test has output: "
cat "$tmpfile.sh"

