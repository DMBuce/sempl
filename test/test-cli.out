Template source is stdin: OK
Template source is -: OK
Template source is arg: OK
Template source is file: OK
Template dest is stdin: OK
Template dest is -: OK
Template dest is file, no output expected:
File from previous test has output: OK
Template dest can be overwritten: STILL OK
Source and dest can't be the same, error expected: '/tmp/sempl.test' and '/tmp/sempl.test' are the same file at ./bin/sempl line 31.
Test $SEMPL_BRACES: OK
Test $LB: {$check}
Test $RB, TODO fix this: {$check{$RB}
Test $SEMPL_DUMP: echo "$check" | while IFS= read -r a; do
echo ''"$a"''
done
Test $SEMPL_DUMP writing to dest, no output expected:
File from previous test has output: exec >"/tmp/sempl.test"
echo "$check" | while IFS= read -r a; do
echo ''"$a"''
done
